config/v1/types_ingress.go

```diff
@@ -172,6 +180,98 @@ type ComponentRouteStatus struct {
+	//   With this policy, if an annotation key in requiredAnnotations is
+	//   not specified on the route at all, the route is rejected.  If every
+	//   annotation key in requiredAnnotations is specified, then the route
+	//   is admitted, irrespective of the annotation values.
+	//
+	// * "AllowWithDefaulting" allows the user to provide arbitrary
+	//   annotation values, and sets defaults if an annotation is missing.
+	//   With this policy, if an annotation key in requiredAnnotations is
+	//   not specified on the route at all, the annotation from
+	//   requiredAnnotations (key and value) is added.
+	//
+	// * "Deny" prohibits the user from specifying annotation values that
+	//   differ from those in requiredAnnotations.  With this policy, if an
+	//   annotation in requiredAnnotations is not specified with the same
+	//   key *and* value on the route, the route is rejected.
+	//
+	// * "Override" overrides user-provided annotations with the annotations
+	//   in requiredAnnotations.  With this policy, if an annotation in
+	//   requiredAnnotations is not specified on the route, the annotation
+	//   is copied from requiredAnnotations is used, and if an annotation
+	//   key in requiredAnnotations is specified on the route, the
+	//   annotation's value is overridden with the value from
+	//   requiredAnnotations.
+	//
+	// Note that the "AllowWithDefaulting" and "Override" options are
+	// dangerous to use.  With these options, a route that is created using
+	// a particular definition one cluster may behave differently from a
+	// route that is created using the same definition but on a cluster with
+	// different required route annotations configured.  Using "Allow" or
+	// "Deny" is safer because either option causes route creation to fail,
+	// loudly, if an annotation is not set as expected for the cluster.
+	Policy RequiredRouteAnnotationsPolicy `json:"policy"`
+
+	// excludedNamespacesSelector may be provided to exempt routes in the
+	// selected namespaces from requiring the annotations in
+	// requiredAnnotations.
+	//
+	// If this field is unset, routes in all namespaces are included.
+	//
+	// +kubebuilder:validation:Optional
+	// +optional
+	ExcludedNamespacesSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// requiredAnnotations is a list of annotations that are required on
+	// newly created routes.  This field's value comprises key-value pairs
+	// where the key specifies a required annotation key and the value
+	// expresses a suggested annotation value.  If a route specifies has an
+	// annotation with each of the required annotation keys, the route is
+	// admitted.  If the route is missing any required annotation, the route
+	// is rejected with a message indicating which annotation key is missing
+	// and what the suggested annotation value is.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	RequiredAnnotations map[string]string `json:"requiredAnnotations"`
```

Comment:
if we narrow the scope to hsts, we can have options in here like 
1. max-age - required exact value, required no longer than, require no shorter than, no opinion
2. preload - required true, required false, no opinion
3. include-subdomains - required true, required false, no opinion